### PR TITLE
feat(BEVFusion): release 2.8.2 

### PR DESCRIPTION
**Pull request by KSeangTan**
## Summary

This PR releases BEVFusion 2.8.2 by fixing and introducing the following features:

- Fix `ObjectRangeMinPointsFilter` by a radius distance [[1]](https://github.com/tier4/AWML/pull/221/changes#diff-67ded712e98b2231ada4e7c2869500bc91df39e4cce866bed372612ff88f9199R89-R91).
- Add corner computation to BEVFusionHead [[2]](https://github.com/tier4/AWML/pull/221/changes#diff-893e7e839dac240e10c5b7ff55d58aee42f5940887d4b7655bcd5f2be310de6dR907-R913).
- Add `RotatedBEVIOULoss` to losses, and set the small weight: `0.10` [[3]](https://github.com/tier4/AWML/pull/221/changes#diff-779f632a83a101b78657ad274672c66a5824a75b9d7bf86676eb1a097572adf6R22).
- Introduce `Gen1_base` [[4]](https://github.com/tier4/AWML/pull/221/changes#diff-f9198806aa6da96ed8c116276f1dca40130911d6ca951b1bca1e6f977e9f51f8R1) and `Gen2_base`, [[5]](https://github.com/tier4/AWML/pull/221/changes#diff-39dc9fe36d4ce6b8318f849968f502015f577358ed4fe8db92eb5cac5cba8912R1) where the new training strategy is now `Gen1_base` -> `Gen2_base` -> `J6gen2_base` or `JPNTaxi_base`.
- Increase training batch size from `16` to `24` [[6]](https://github.com/tier4/AWML/pull/221/changes#diff-56163bf7ac53126d9b7dbd04491d509c1417b09b26b1c965d6f843e70e47e3bbR9).
- Increase training epochs from `30` to `50` for `j6gen2_base` [[7]](https://github.com/tier4/AWML/pull/221/changes#diff-2ea3181bc7f56642d008057b66514bece13caf9c3f709f8519727246a85c4bd8R6).

### LargeBus
ap_center_distance_bev_largebus

---

### J6Gen2

ap_center_distance_bev_j6gen2

---

### J6Gen2_Base
ap_center_distance_bev_j6gen2_base

---

### JPNTaxi_Gen2
ap_center_distance_bev_jpntaxi_gen2

---

### CODE REVIEW

File: autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py [hunk 1]
```diff
@@ -85,17 +86,9 @@ def transform(self, input_dict: dict) -> dict:
...
-
-        # ---- lower/upper bound for radius ----
-        lower_mask = gt_bboxes_3d.in_range_bev(lower_bev_range)  # e.g. min_radius = 5m
-        upper_mask = gt_bboxes_3d.in_range_bev(upper_bev_range)  # e.g. max_radius = 70m
-        bev_radius_mask = (lower_mask | upper_mask).numpy().astype(bool)  # final mask
-
-        # Out of range gt masks are all valid
+        # Get radius of each bev center
+        bbox_bev_centers = np.linalg.norm(gt_bboxes_3d.bev[:, :2], axis=1)
+        bev_radius_mask = (bbox_bev_centers >= self.range_radius[0]) and (bbox_bev_centers < self.range_radius[1])
```
Comment (vividf) [inline autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py:91]: bbox_bev_centers is a NumPy array, so and will try to evaluate the truth value of the entire array and raise ValueError. Please use element-wise & instead:
```python
bev_radius_mask = (
    (bbox_bev_centers >= self.range_radius[0])
    & (bbox_bev_centers < self.range_radius[1])
)
```

File: autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py [hunk 2]
```diff
@@ -85,17 +86,9 @@ def transform(self, input_dict: dict) -> dict:
...
-        upper_bev_range = [self.range_radius[0], self.range_radius[0], self.range_radius[1], self.range_radius[1]]
-
-        # ---- lower/upper bound for radius ----
-        lower_mask = gt_bboxes_3d.in_range_bev(lower_bev_range)  # e.g. min_radius = 5m
-        upper_mask = gt_bboxes_3d.in_range_bev(upper_bev_range)  # e.g. max_radius = 70m
-        bev_radius_mask = (lower_mask | upper_mask).numpy().astype(bool)  # final mask
-
-        # Out of range gt masks are all valid
+        # Get radius of each bev center
+        bbox_bev_centers = np.linalg.norm(gt_bboxes_3d.bev[:, :2], axis=1)
```
Comment (vividf) [inline autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py:90]: nit: bbox_bev_center_distances or bbox_bev_center_radii is better (?)

File: autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py
Comment (vividf) [inline autoware_ml/detection3d/datasets/transforms/object_min_points_filter.py:59]: The docstring still describes the old square-range behaviour ("This will be transformed to [-max, -max, -min, -min] and [min, min, max, max]"). Since the semantics changed to a Euclidean radius, could you update it here too?

File: projects/BEVFusion/bevfusion/bevfusion_head.py [hunk 3]
```diff
@@ -873,6 +877,16 @@ def loss_by_feat(
                 preds = torch.cat([layer_center, layer_height, layer_dim, layer_rot, layer_vel], dim=1).permute(
                     0, 2, 1
                 )  # [BS, num_proposals, code_size]
+            
+            if "iou_preds" in preds_dict.keys():
+                layer_iou_preds = preds_dict["iou_preds"][
```
Comment (vividf) [inline projects/BEVFusion/bevfusion/bevfusion_head.py:882]: `layer_iou_preds` is computed here and then never used, and `self.loss_iou_preds(...)` is never called anywhere in the file. So if someone configures loss_iou_preds, the iou_preds branch takes part in forward but receives no gradient — while predict_by_feat (line 421) still rescales scores with sigmoid(iou_preds). That means inference would be scaled by weights that never left their initialisation.

Two options:

(a) finish it: add loss_dict[f"{prefix}_loss_iou_preds"] = self.loss_iou_preds(layer_iou_preds, ious.detach(), layer_iou_weights, avg_factor=...), add iou_preds=[1, 2] to common_heads, and confirm the ONNX export and the Autoware side can consume the extra output.
(b) drop it from this PR: remove loss_iou_preds, layer_iou_preds, and the predict_by_feat branch, and land it separately.
